### ISSUE TITLE
Keyboard navigation is not disabled when grid cell contains an input

### ISSUE THREAD

**Issue by LoGA80** [vmware-clarity/ng-clarity#473]
## Describe the bug

We have been using Clarity Angular components in our enterprise application and keeping them as up-to-date as possible (currently on Angular 13 base). A user just reported an issue about not being able to scroll inside an edit control that is in a Clarity Datagrid. Investigation lead to a finding that the same issue occurs for each and every of our datagrids that contains input (=text) controls. We have so far had not found a general solution or configuration option for the Clarity control.

## How to reproduce

Create a datagrid with a couple of rows and columns, at least one of the cells have to contain an input like this:
`<clr-dg-cell>
<input ...>
</clr-dg-cell>`

When using the keyboard arrow keys to cycle through the data in the grid, it will not allow to move the cursor inside the input.

## Expected behavior

When entering to a cell with an input control, the keyboard navigation should be handed over to the input control so that scrolling within the text becomes possible.

## Versions

**Clarity project:**

- [ ] Clarity Core
- [X] Clarity Angular/UI

**Clarity version:**

- [ ] v5.x
- [X] v6.x

**Framework:**

- [X] Angular
- [ ] React
- [ ] Vue
- [ ] Other:

**Framework version:**
Angular 13

**Device:**

- Type: Windows PC
- OS: Windows 10
- Browser Chrome
- Version 108.0.5359.125

## Additional notes

The workaround we have been using to fix the problem is to hijack the keydown event and stop propagating when the arrow keys are used:

`preventInputJump(event) {
  if (event.key === 'ArrowLeft' || event.key === 'ArrowRight') {
    event.stopImmediatePropagation();
  }
}`

**Comment by LoGA80** [comment]
We have a couple of grids with input fields, isn't there a plan to add a general solution? Even a grid-level "disable keyboard navigation" would be an option for us.

**Comment by Jinnie** [comment]
It is all about Accessibility.
Our Accessibility team recommendation is to either activate any editable field and deactivate it after edit (Like put "Edit" link next to it) or make the whole record (row) editable in a separate dialog, for example in a datagrid detail panel.
Having open input fields inside the datagrid is not acceptable from A11Y point of view.
The keyboard navigation is also an important accessibility feature, which we do not consider to be optional.
Technically, you _can_ prevent the key navigation. The demo Kevin shared above also features a directive, which can easily be modified to work without parameters and be applied globally to any "input" field, so you don't need to apply to to each place they are used, but this would lead to Accessibility violations, which can not be supported by Clarity OOTB.

**Comment by LoGA80** [comment]
Thanks for the answers, I understand but am a bit disappointed. We will implement our own solution to overcome the problem.
